stdlib/public/core/MutableCollection.swift

```diff
@@ -398,6 +398,29 @@ extension MutableCollection where Self: RandomAccessCollection {
   }
 }
 
+extension MutableCollection {
+  /// Exchange the values at indices `i` and `j`.
+  ///
+  /// Has no effect when `i` and `j` are equal.
+  @_inlineable
+  public mutating func swapAt(_ i: Index, _ j: Index) {
+    guard i != j else { return }
+    
+    // Semantically equivalent to (i, j) = (j, i).
+    // Microoptimized to avoid retain/release traffic.
+    let p1 = Builtin.addressof(&self[i])
+    let p2 = Builtin.addressof(&self[j])
```

Comment:
And as far as the benchmarks are concerned, replacing the optimized collection with a much simpler one makes no material difference. Though maybe it would with more targeted benchmarks.